### ISSUE TITLE
Up our table layout game

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#4621]
I think the core team was all pretty keen on [this article](https://medium.com/mission-log/design-better-data-tables-430a30a00d8c#.rlq1ic1ih), and it made a few obvious suggestions that I think we should implement in our tables. Namely:

- [x] Numerical data should be right-aligned (!!!!)
- [x] Textual data should be left-aligned
- [x] Headers should be aligned with their data (i.e., headings for numerical fields should be right-aligned)
- [x] 3½. Don’t use center alignment.

I think we'll want to make exceptions for category fields that happen to be numerical — e.g., I don't think a phone number or ZIP code should be right-aligned.

It seems our typeface has pretty good, even spacing for numbers, so I think we're okay there but would be good to take another look to make sure things like decimal points and commas line up as they should.

**Comment by tlrobinson** [comment]
For a second I thought you wanted us to port our HTML markup to use `<tables>` for layout...

**Comment by mazameli** [comment]
Well, it _would_ improve cross-browser consistency… :trollface:

**Comment by mazameli** [comment]
I also think we could experiment with getting rid of our vertical borders, slightly tightening the spacing in between rows, and slightly decreasing the font sizing. One wrinkle with this would be figuring out a way to indicate in the UI that you can resize the width of columns.

![table-redesign](https://cloud.githubusercontent.com/assets/2223916/24424952/325e8a2c-13b7-11e7-8695-4d45b798f25e.png)

**Comment by salsakran** [comment]
really like it visually. additionally, remember that table headings will have actions once #4496 is merged.

**Comment by tlrobinson** [comment]
@mazameli how might pivoted tables look?

**Comment by mazameli** [comment]
Yeah, as far as table heading actions, I think we can make the headings go blue on hover. (As an aside we'll have to figure out if sort-by-this-column is an action within the other actions menu, or if it's its own thing. I'm thinking the former right now.)

Pivot tables (i.e., 1 metric 2 dimension pivot tables) could look like this:

![pivot table v1](https://cloud.githubusercontent.com/assets/2223916/24671799/aa7a5a52-1927-11e7-8e29-eb4ed649483e.png)

**Comment by kdoh** [comment]
Generally 👍 . Only thing that could get a bit odd in the case of the pivot example you posted is the alignment of the "x axis" pivoted field label and the right aligned labels for the different values. In the screenshot it's alignment with Apple is almost but not quite and it's wigging me out :P

**Comment by mazameli** [comment]
Yeah, I waffled back and forth a bunch on where to align that label (`Filling` in the mock). Maybe we make it have the same alignment as the first heading in the x-axis, like so:

![pivot table v1](https://cloud.githubusercontent.com/assets/2223916/24680253/ff145c9e-1944-11e7-8cbb-efb9327e9556.png)

**Comment by tlrobinson** [comment]
Cool.

There was also the summary column/row in the new query builder design. I imagine they'd just look like the other row/column, but maybe fixed so they're always visible without scrolling?

**Comment by mazameli** [comment]
Haven't thought that far TBH. In this issue I was trying to just propose CSS changes, and I see the subtotal/total cells as being a new feature. Good question though.

**Comment by tlrobinson** [comment]
Resolved by #4977

**Comment by mazameli** [comment]
I'll open a separate issue about stripping down the extraneous grid lines.

**Comment by EvaSchreyer** [comment]
Hi, I use metabase v0.25 and the alignment of colums is still not great:
![image](https://user-images.githubusercontent.com/28557318/30055721-9c688150-9230-11e7-97f5-6b439da67a3f.png)

As you can see I concat some of the numbers to add the "EUR" as there is currently no way to a) change the decimal dot to comma and b) to add a currency sign. Therefore the data is a string and is not aligned like the numbers in the other columns.

**Comment by EvaSchreyer** [comment]
Hi again, in case you still define the adjustments for the change it would be great if the user/the admin could decide if a column is right or left aligned. 

The proposal:
Numerical data should be right-aligned (!!!!)
Textual data should be left-aligned

doesn´t really work when you still need concat syntax to add currency signs to your data - as then the number is turned into a string.

**Comment by cqcn1991** [comment]
If the alignment problem is solved? Seems still not.

**Comment by flamber** [comment]
@cqcn1991 Which version of Metabase? Can you provide a screenshot of your problem or more details?

**Comment by Dcushing57** [comment]
Hi,

It does not appear the the pivot table headers do not use right alignment when value's data type is Number. I am also using 3 category data types, 2 in the column, 1 in the row. Neither header is right aligned.

I received a complaint that the pivot table is hard to read as it almost looks like things are misaligned due to the left alignment on both headers and right alignment on number.

I am using a pivot table with the following details. I'm sorry, I cannot provide screenshot due to PHI. I will try to describe details as close as I can.

Summarization:

Sum(Number data type) by Category 1, Category 2, Category 3

Rows:
Category 1

Columns:
Category 2
Category 3

Values:
Sum(Number)

**Comment by flamber** [comment]
@Dcushing57 This issue is closed and several years old. You are seeing this issue: #16752
